contracts/src/v0.8/dev/transmission/entrypoint_implementation/contracts/SCALibrary.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+library SCALibrary {
+  // keccak256("EIP712Domain(uint256 chainId, address verifyingContract)");
+  bytes32 internal constant DOMAIN_SEPARATOR = hex"1c7d3b72b37a35523e273aaadd7b4cd66f618bb81429ab053412d51f50ccea61";
+
+  // keccak256("executeTransactionFromEntryPoint(address to, uint256 value, bytes calldata data)");
+  bytes32 internal constant TYPEHASH = hex"4750045d47fce615521b32cee713ff8db50147e98aec5ca94926b52651ca3fa0";
+
+  enum LinkPaymentType {
+    DIRECT_FUNDING,
+    SUBSCRIPTION // TODO: implement
+  }
+
+  struct DirectFundingData {
+    address recipient; // recipient of the top-up
+    uint256 topupThreshold; // set to zero to disable auto-topup
+    uint256 topupAmount;
+  }
+
+  function getUserOpFullHash(bytes32 userOpHash) internal pure returns (bytes32 fullHash) {
+    bytes32 hashOfEncoding = keccak256(abi.encode(SCALibrary.TYPEHASH, userOpHash));
+    fullHash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0x01), SCALibrary.DOMAIN_SEPARATOR, hashOfEncoding));
+  }
```

Comment:
Is this a more-or-less standard implementation? I'm kind of curious because I thought that the `chainId` should be part of the domain separator (following EIP-712) but it seems like the `DOMAIN_SEPARATOR` here is the same for all chains, which seems to make it susceptible to replay attacks. Is this replay attack handled somewhere else?

Reply:
Wow I've inadvertently removed chainID from the signature check. Need to reincorporate it.